linalg/solvers.cpp

```diff
@@ -103,6 +104,49 @@ void IterativeSolver::SetOperator(const Operator &op)
 }
 
 
+OperatorJacobiSmoother::OperatorJacobiSmoother(const Vector &d,
+                                               const Array<int> &ess_tdofs,
+                                               const double dmpng)
+   :
+   Solver(d.Size()),
+   N(d.Size()),
+   dinv(N),
+   diag(d),
+   damping(dmpng),
+   ess_tdof_list(ess_tdofs),
+   residual(N) { Setup(); }
+
+void OperatorJacobiSmoother::Setup()
+{
+   residual.UseDevice(true);
+   const double delta = damping;
+   auto D = diag.Read();
+   auto X = dinv.Write();
+   MFEM_FORALL(i, N, X[i] = delta / D[i]; );
+   auto I = ess_tdof_list.Read();
+   MFEM_FORALL(i, ess_tdof_list.Size(), X[I[i]] = delta; );
+}
+
+void OperatorJacobiSmoother::Mult(const Vector &x, Vector &y) const
+{
+   if (iterative_mode && oper)
+   {
+      oper->Mult(y, residual);  // r = A x
+      subtract(x, residual, residual); // r = b - A x
+   }
+   else
+   {
+      residual = x;
+      y.UseDevice(true);
+      y = 0.0;
+   }
+   auto DI = dinv.Read();
```

Comment:
This variable is called `X` in the previous function, we should either change `DI` to `X` or `X` to `DI` I think.

Reply:
Done.